core/templates/dev/head/components/create_button/CreateActivityButtonDirective.ts

```diff
@@ -32,20 +35,20 @@ oppia.directive('createActivityButton', [
             ExplorationCreationService, CollectionCreationService,
             SiteAnalyticsService, UrlService, UserService,
             ALLOW_YAML_FILE_UPLOAD) {
-          $scope.creationInProgress = false;
-          $scope.allowYamlFileUpload = ALLOW_YAML_FILE_UPLOAD;
+          this.creationInProgress = false;
+          this.allowYamlFileUpload = ALLOW_YAML_FILE_UPLOAD;
 
-          $scope.canCreateCollections = null;
-          $scope.userIsLoggedIn = null;
+          this.canCreateCollections = null;
+          this.userIsLoggedIn = null;
           UserService.getUserInfoAsync().then(function(userInfo) {
-            $scope.canCreateCollections = userInfo.canCreateCollections();
-            $scope.userIsLoggedIn = userInfo.isLoggedIn();
+            this.canCreateCollections = userInfo.canCreateCollections();
```

Comment:
Can you check that this is updating the correct thing? I think `this` refers to something different when inside an anonymous function.